### PR TITLE
Cleanup DotvvmConfiguration serialization format

### PR DESCRIPTION
**Pull request by exyi**
Esp. from the `$type: FreezableList`1[...]`, but also from other clutter.

Added manual acceptance tests for that, so we can maintain the format
in the future.

@Mylan719 Would you please check that the examples (in the JSON files) match your expectation?

### CODE REVIEW

**Comment by quigamdev** [comment]
@exyi I have just checked all sections of the generated json file. It seems good to me.

File: src/DotVVM.Framework.Tests.Common/Runtime/ConfigurationSerializationTests.cs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+            c.RouteTable.Add("route1", "url1", "file1.dothtml", new { a = "ccc" });
+            c.RouteTable.Add("route2", "url2/{int}", "file1.dothtml", new { a = "ccc" });
+            c.RouteTable.Add("custom presenter", "url3", "", presenterFactory: LocalizablePresenter.BasedOnQuery("lang"));
+
+            c.RouteTable.AddGroup("group1", "group-{lang}/", "", rg => {
+
+                rg.Add("r1", "r1", "g-r1.dothtml");
+
+                rg.AddGroup("g", "g/", "", rg => {
```
Comment (quigamdev) [inline src/DotVVM.Framework.Tests.Common/Runtime/ConfigurationSerializationTests.cs:50]: This is a compilation error. `rg.AddGroup("g", "g/", "", rg => {` there is declaration of existing variable rg. @exyi

Comment (exyi) [reply src/DotVVM.Framework.Tests.Common/Runtime/ConfigurationSerializationTests.cs:50]: This should be valid in modern C#

File: src/DotVVM.Framework.Tests.Common/Runtime/config-tests/ConfigurationSerializationTests.Markup.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      },
+      {
+        "namespace": "System.Collections.Generic"
+      },
+      {
+        "namespace": "System.Collections",
+        "alias": "Collections"
+      }
+    ],
+    "defaultExtensionParameters": [
```
[the remark this answers is not in the document]
Comment (exyi) [reply src/DotVVM.Framework.Tests.Common/Runtime/config-tests/ConfigurationSerializationTests.Markup.json:36]: Yes it was, I wanted to be consistent in the format.

File: src/DotVVM.Framework.Tests.Common/Runtime/ConfigurationSerializationTests.cs (hunk 1 above)
Comment (quigamdev) [reply src/DotVVM.Framework.Tests.Common/Runtime/ConfigurationSerializationTests.cs:50]: I see. I don't know why but my VS was not able to build it. Now it works.

**Pull request merged**